pkg/clusteragent/autoscaling/workload/pod_watcher.go

```diff
@@ -27,30 +31,34 @@ type NamespacedPodOwner struct {
 	Name string
 }
 
-type podWatcher struct {
-	mutex           sync.RWMutex
-	wlm             workloadmeta.Component
-	podsPerPodOwner map[NamespacedPodOwner]map[string]*workloadmeta.KubernetesPod
-}
-
-// PodWatcher indexes pods by their owner
-type PodWatcher interface {
+// podWatcher indexes pods by their owner
+type podWatcher interface {
 	// Start starts the PodWatcher.
-	Start(ctx context.Context)
+	Run(ctx context.Context)
 	// GetPodsForOwner returns the pods for the given owner.
 	GetPodsForOwner(NamespacedPodOwner) []*workloadmeta.KubernetesPod
 }
 
+type podWatcherImpl struct {
+	mutex sync.RWMutex
+
+	wlm             workloadmeta.Component
+	patcher         PodPatcher
+	patcherChan     chan *workloadmeta.KubernetesPod
```

Comment:
I think the channel doesn't have to be provided. We can have a method in the PodPatcher